src/ndisrcdemux/imp.rs

```diff
@@ -276,6 +276,10 @@ impl NdiSrcDemux {
 
         if add_pad {
             element.add_pad(&srcpad).unwrap();
+            if element.num_src_pads() == 2 {
```

Comment:
I would go via `let no_more_pads = add_pad && state.video_pad.is_some() && state.audio_pad.is_some()` but seems fine to me otherwise